src/abstractdataframe/abstractdataframe.jl

```diff
@@ -981,43 +997,91 @@ julia> vcat(df1, df2)
 │ 4   │ 4     │ 4     │
 │ 5   │ 5     │ 5     │
 │ 6   │ 6     │ 6     │
+
+julia> vcat(df1, df3, columns = :union)
+6×3 DataFrame
+│ Row │ A     │ B       │ C       │
+│     │ Int64 │ Int64⍰  │ Int64⍰  │
+├─────┼───────┼─────────┼─────────┤
+│ 1   │ 1     │ 1       │ missing │
+│ 2   │ 2     │ 2       │ missing │
+│ 3   │ 3     │ 3       │ missing │
+│ 4   │ 7     │ missing │ 7       │
+│ 5   │ 8     │ missing │ 8       │
+│ 6   │ 9     │ missing │ 9       │
+
+vcat(df1, df3, columns = :intersect)
+6×1 DataFrame
+│ Row │ A     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+│ 3   │ 3     │
+│ 4   │ 7     │
+│ 5   │ 8     │
+│ 6   │ 9     │
 ```
 """
-Base.vcat(df::AbstractDataFrame) = df
```

Comment:
I've removed this method since it doesn't make a copy, which is inconsistent with the general case (and can be dangerous). We could write an optimized implementation but I guess it doesn't matter a lot.

Reply:
Good point. I'm going to add a method for an empty dataframe, though.

Reply:
The ownership PR sets this to `Base.vcat(df::AbstractDataFrame) = DataFrame(df)` and I think this is how it could stay (but a general `vcat` should handle it correctly - just please make sure that we test this case).

Reply:
So what's the best solution? Which PR should we merge first?